### DISCUSSION TITLE
Pytorch fundamentals: "Why is this not a tensor?"

### DISCUSSION THREAD

**Discussion by kkdotpy** [mrdbourke/pytorch-deep-learning#959] [Q&A] [answered]
Hello, I am new to pytorch so I do not have much idea about tensors. I tried creating a 6 dimensional tensor so as to understand how the dimension and shape command work. However, I noticed that the following tensor is returning some sort of error 

```python 
ten_6 = torch.tensor([[[[[[1,1],[1,1]],[[1,1],[1]],[[1,1],[1,1]],[[1,1],[2,2]]]]]])
print(ten_6)
print(ten_6.shape)
print(ten_6.ndim)
```
# Error 
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/144082123/009ebea5-9cc3-4c26-822c-d22cd5bebdbc)

Meanwhile this tensor is just fine 

``` python 
ten_6 = torch.tensor([[[[[[1,1],[1,1]],[[1,1],[1,1]],[[1,1],[1,1]],[[1,1],[2,2]]]]]])
print(ten_6)
print(ten_6.shape)
print(ten_6.ndim)
```
Can someone please help me understand this bit? Thank you..

**Comment by matrx2000** [comment, marked as answer]
Tensors need to have consistent dimensions at each level of nesting. In the provided list structure, all sub-lists at the same level must have the same number of elements. In your example:

```python
[
    [
        [
            [
                [
                    [1,1],[1,1]
                ],
                [
                    [1,1],[1] #<-- error is here
                ],
                [
                    [1,1],[1,1]
                ],
                [
                    [1,1],[2,2]
                ]
            ]
        ]
    ]
]
```
